mlrun/api/schemas/model_endpoints.py

```diff
@@ -22,73 +28,225 @@ class ModelEndpointSpec(ObjectSpec):
+                histogram=Histogram(buckets=stats["hist"][1], counts=stats["hist"][0]),
+            )
+        else:
+            return None
+
 
 class Features(BaseModel):
     name: str
     weight: float
-    expected: FeatureValues
+    expected: Optional[FeatureValues]
     actual: Optional[FeatureValues]
 
+    @classmethod
+    def new(
+        cls,
+        feature_name: str,
+        feature_stats: Optional[dict],
+        current_stats: Optional[dict],
+    ):
+        return cls(
+            name=feature_name,
+            weight=-1.0,
+            expected=FeatureValues.new(feature_stats),
+            actual=FeatureValues.new(current_stats),
+        )
+
 
 class ModelEndpointState(BaseModel):
     endpoint: ModelEndpoint
-    first_request: Optional[str] = None
-    last_request: Optional[str] = None
-    accuracy: Optional[float] = None
-    error_count: Optional[int] = None
-    drift_status: Optional[str] = None
-    metrics: Dict[str, Metric] = {}
-    features: List[Features] = []
+    first_request: Optional[str]
+    last_request: Optional[str]
+    accuracy: Optional[float]
+    error_count: Optional[int]
+    drift_status: Optional[str]
+    drift_measures: Optional[dict]
+    current_stats: Optional[dict]
+    metrics: Optional[Dict[str, Metric]]
+    features: Optional[List[Features]]
+
+    @classmethod
+    def new(
+        cls,
+        endpoint: ModelEndpoint,
+        first_request: Optional[str] = None,
+        last_request: Optional[str] = None,
+        accuracy: Optional[float] = None,
+        error_count: Optional[int] = None,
+        drift_status: Optional[str] = None,
+        drift_measures: Optional[dict] = None,
+        current_stats: Optional[dict] = None,
+        metrics: Optional[Dict[str, Metric]] = None,
+        features: Optional[List[Features]] = None,
+    ):
+        """ A constructor method for better usability
+
+        :param endpoint: The object representing a model endpoint
+        :param first_request: The first request encountered for the given endpoint
+        :param last_request: The last request encountered for the given endpoint
+        :param accuracy: The accuracy of the model
+        :param error_count: The error count of the model
+        :param drift_status: The drift status of the model
+        :param drift_measures: The drift measures of the model
+        :param current_stats: The current statistics of the model
+        :param metrics: The collected metrics of the model endpoint
+        :param features: The incoming features of the model
+        """
+        return cls(
+            endpoint=endpoint,
+            first_request=first_request,
+            last_request=last_request,
+            accuracy=accuracy,
+            error_count=error_count,
+            drift_status=drift_status,
+            drift_measures=drift_measures,
+            current_stats=current_stats,
+            metrics=metrics,
+            features=features,
+        )
```

Comment:
What's the purpose here ? why not using the regular constructor ?

Reply:
Because the regular constructor does not provide me with actual IDE support. the regular constructor requires `*args` and `**kwargs`, creating the internal structure of this object every time i need to construct it is very verbose. I want to support both `**kwargs` style unpacking, and a more structured way of initializing the object.